experimental/web/index.html

```diff
@@ -0,0 +1,152 @@
+<!DOCTYPE html>
+<html>
+
+<head>
+    <meta charset="utf-8" />
+    <title>Oak emulator</title>
+</head>
+
+<body>
+    <script src="https://cdn.jsdelivr.net/npm/vue/dist/vue.js"></script>
+    <script type="module">
+        function init() {
+            var app = new Vue({
+                el: '#app',
+                data: {
+                    // Exports of the loaded Module.
+                    exports: [],
+                    // Import of the loaded Module.
+                    imports: [],
+                    // Oak ABI calls trace.
+                    trace: [],
+                    // Loaded module.
+                    module: null,
+                    // Loaded module instance.
+                    instance: null,
+                },
+                methods: {
+                    // Read the specified file and load it as a Wasm module.
+                    readFile: function (e) {
+                        const file = e.target.files[0];
+                        if (!file) {
+                            console.log('no file selected');
+                            return;
+                        }
+                        const reader = new FileReader();
+                        reader.onload = this.loadModule;
+                        reader.readAsArrayBuffer(file);
+                    },
+                    // Parse the provided ArrayBuffer as a Wasm module and load it as an Oak module,
+                    // providing the necessary imports.
+                    loadModule: async function (e) {
+                        const contents = e.target.result
+                        console.log('file loaded');
+                        this.module = await WebAssembly.compile(contents);
+                        this.exports = WebAssembly.Module.exports(this.module);
+                        this.imports = WebAssembly.Module.imports(this.module);
+                        // Bind this to a variable so that it can be referred to from the
+                        // implementation of the Oak ABI functions.
+                        const v = this;
+                        const statusOk = 1;
+                        // Provide a mock implementation of some of the Oak ABI functions.
+                        // Mostly these just log their argument to the trace, and return a
+                        // successful status without actually doing anything.
+                        const importObject = {
+                            oak: {
+                                wait_on_channels: (buf, count) => {
+                                    const ret = statusOk;
```

Comment:
What is `ret` a shorthand for?

Reply:
"return", but happy to change to something else if there is any convention?

Reply:
Oh that makes sense. :) Afaik there isn't a JS convention, but I'd suggest `result` (we often use that in the Rust runtime) or maybe `returnValue`. 

My personal nitty opinion, but imho, except for exceedingly common cases, the time savings of abbreviated var names rarely justify friction introduced when reading them.

Reply:
renamed to `status`, I think here it makes sense to be more specific :)